pkg/ipam/ipallocator/allocator.go

```diff
@@ -138,30 +142,84 @@ func (a *SingleIPAllocator) AllocateIP(ip net.IP) error {
 	return nil
 }
 
+func (a *SingleIPAllocator) allocateOffset(i int) (net.IP, bool) {
+	if a.allocated.Bit(i) == 0 {
+		ip := utilnet.AddIPOffset(a.base, i)
+		if a.checkReserved(ip) != nil {
+			return nil, false
+		}
+		a.allocated.SetBit(a.allocated, i, 1)
+		a.count++
+		return ip, true
+	}
+
+	return nil, false
+}
+
 // AllocateNext allocates an IP from the IP range. It returns error if no IP is available.
 func (a *SingleIPAllocator) AllocateNext() (net.IP, error) {
 	a.mutex.Lock()
 	defer a.mutex.Unlock()
 	if a.count >= (a.max - len(a.reservedIPs)) {
 		return nil, fmt.Errorf("no available IP")
 	}
-	for i := 0; i < a.max; i++ {
-		if a.allocated.Bit(i) == 0 {
-			ip := utilnet.AddIPOffset(a.base, i)
-			if a.checkReserved(ip) != nil {
-				continue
-			}
-			a.allocated.SetBit(a.allocated, i, 1)
-			a.count++
+	for i := 0; i <= a.max; i++ {
+		if ip, ok := a.allocateOffset(i); ok {
 			return ip, nil
 		}
 	}
+
+	// we should never reach here
 	return nil, fmt.Errorf("no available IP")
 }
 
+// Allocate continuous range of specified size. If not available, error is returned.
```

Comment:
```suggestion
// AllocateRange allocates continuous range of specified size. If not available, error is returned.
```
To align with other comments

Reply:
Done